src/worker.py

```
@@ -7101,6 +7107,162 @@ async def api_get_thread_messages(req, env, thread_id: str):
+        return err("Image must be 5 MB or smaller")
+
+    r2 = getattr(env, "R2", None)
+    r2_key = ""
+
+    if r2:
+        # Upload to R2 and store the public URL
+        ext = image_type.split("/")[-1].replace("jpeg", "jpg")
+        r2_key = f"avatars/{user['id']}/{new_id()}.{ext}"
+        try:
+            await r2.put(r2_key, to_js(img_bytes))
+        except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)